### DISCUSSION TITLE
Normalization and gene matrix

### DISCUSSION THREAD

**Discussion by AnjaliC4** [stuart-lab/signac#507] [Q&A] [answered]
Hi, I have two independent questions

1. I think this may have been  discussed before but Iwas wondering what is the consensus on merging and normalizing scATAC-seq data. Is it recommended to normalize every subject after creating a new feature matrix with consensus peaks and then merging them or merging followed by normalization with TF-IDF. Also, could you please explain how the two would differ in terms of results?
In one of my trials, I normalized every feature matrix with consensus peaks and then merged them followed by re-normalization and integration which for some reason gave more separated clusters than  simply merging and normalizing. 

2. In addition, I see that I have XIST promoter peaks in my data, but then I don't get XIST after creating gene.matrix from ATAC. XIST promoter falls right at the end of gene sequence. I know you flank peaks +/- 1kbp so it should be covered, but it does not show up. If this may be because it has perhaps low expression, even then it should be in the matrix since a promoter peak is present in ATAC data.  

Please let me know your thoughts. Thanks!

**Comment by timoast** [comment, marked as answer]
For (1), please see https://github.com/timoast/signac/discussions/469

For (2), XIST is not a protein-coding gene and only protein-coding genes are used in the `GeneActivity()` function. If you want to include non-protein-coding genes, you should supply your own set of genomic ranges to quantify to the `FeatureMatrix()` function.

**Reply by AnjaliC4** [reply]
Hi Tim, thanks, I see you referred to discuss 469, my understanding is that it does not matter if one does normalization before or after merging, but I see differences in the clustering with both techniques. Moreover, does it make sense to normalize each seurat object, merge them and then normalize the merge object again? Please clarify.

For non-coding genes, is it possible to add a tutorial. The genomic ranges that go in my featureMatrix are the peaks called by MACS2, so I am not sure how can I tweak that to include non-coding genes. 
I am fairly new to these analysis, so I would really appreciate a bit more clarifications on normalization and steps to include non-coding genes along with coding. Thank you!

**Reply by timoast** [reply]
If you normalised the data, merged, then normalised again, the second normalisation will just overwrite the first so it won't make any difference if you ran it before merging or not.

FeatureMatrix will quantify the number of Tn5 insertions within any given set of genomic ranges. In the examples on the website we typically use it to quantify peaks, but you can use it for any genomic range. If you wanted to quantify the region around XIST, you just need to supply those genomic coordinates. I don't have a lot of bandwidth at the moment to create new tutorials, but it's something we could think about adding in the future.

**Reply by AnjaliC4** [reply]
Thanks! Got the first one.

For second, I still do not fully follow. I already have genomic range mapping to the promoter of XIST gene in my ATAC matrices, please see the image below. They just do not show up in gene_activity matrix. Is there a way to add a module to include lncRNA as well in the gene.activity?

![image](https://user-images.githubusercontent.com/60046859/110859069-dffd1380-8288-11eb-8fa9-bd888f785a3c.png)

**Reply by timoast** [reply]
The `GeneActivity()` function is independent from the peak calling you've done, and so whether there is a peak at the XIST gene or not does not affect anything. The thing that determines whether the gene is included in the output is if it is a protein coding gene or not. In the case of XIST, it's noncoding and so it's not included. There's currently no option in `GeneActivity()` to include noncoding genes. However, all `GeneActivity()` does is extract the gene coordinates and quantify those regions using the `FeatureMatrix()` function. To find the total number of Tn5 insertions at the XIST gene ("gene activity"), you can supply the genomic ranges for the XIST gene to `FeatureMatrix()`. If you want to include all noncoding transcripts in the gene activity assay, you'd need to extract the genomic coordinates for all of those genes and supply them to the `FeatureMatrix()` function.

**Reply by AnjaliC4** [reply]
Ok got it, thanks a lot!!
